lib/std/Build/Step/Run.zig

```diff
@@ -588,12 +587,55 @@ fn make(step: *Step, prog_node: *std.Progress.Node) !void {
         argv_list.items[placeholder.index] = cli_arg;
     }
 
-    try runCommand(self, argv_list.items, has_side_effects, &digest, prog_node);
+    try runCommand(self, argv_list.items, has_side_effects, tmp_dir_path, prog_node);
 
     if (self.dep_output_file) |dep_output_file|
         try man.addDepFilePost(std.fs.cwd(), dep_output_file.generated_file.getPath());
 
+    const digest = man.final();
+
+    // rename in place
+    const o_sub_path = try std.fs.path.join(arena, &[_][]const u8{ "o", &digest });
+    try b.cache_root.handle.makePath(o_sub_path);
+    try b.cache_root.handle.rename(tmp_dir_path, o_sub_path);
```

Comment:
Catch both of these errors and report an error, just like you see above with the call to `b.cache_root.handle.makePath(`. This will likely make the CI failure easier to figure out (as well as a better error message for every zig build user).